proc_macros/src/lib.rs

```diff
@@ -44,3 +45,108 @@ pub fn builder_args(input: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+struct InvalidRuleTestSpec {
+    code: Expr,
+    errors: ExprArray,
+}
+
+impl Parse for InvalidRuleTestSpec {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let mut code: Option<Expr> = Default::default();
```

Comment:
It might be interesting to try writing a "meta-proc-macro" for doing this "parse the expected `key => whatever` syntax for an expected set of keys + corresponding value "parse-types""?